nixnet/_session/frames.py

```diff
@@ -30,14 +30,43 @@ def read_bytes(
             self,
             number_of_bytes_to_read,
             timeout=constants.TIMEOUT_NONE):
-        """Read frames.
-
-        Valid modes
-        - Frame Input Stream Mode
-        - Frame Input Queued Mode
-        - Frame Input Single-Point Mode
-        Frame: one or more
-        http://zone.ni.com/reference/en-XX/help/372841N-01/nixnet/nxreadframe/
+        """Read data as a list of raw bytes (frame data).
+
+        The raw bytes encode one or more frames using the Raw Frame Format.
+
+        Args:
+            number_of_bytes_to_read: An integer repesenting the number of bytes to read.
+            timeout: The time to wait for number to read frame bytes to become
+                available; the 'timeout' is represented as 64-bit floating-point
+                in units of seconds.
+
+                To avoid returning a partial frame, even when
+                'number_of_bytes_to_read' are available from the hardware, this
+                read may return fewer bytes in buffer. For example, assume you
+                pass 'number_of_bytes_to_read' 70 bytes and 'timeout' of 10
+                seconds. During the read, two frames are received, the first 24
+                bytes in size, and the second 56 bytes in size, for a total of
+                80 bytes. The read returns after the two frames are received,
+                but only the first frame is copied to data. If the read copied
+                46 bytes of the second frame (up to the limit of 70), that frame
+                would be incomplete and therefore difficult to interpret. To
+                avoid this problem, the read always returns complete frames in
+                buffer.
+
+                If 'timeout' is positive, this function waits for
+                'number_of_bytes_to_read' frame bytes to be received, then
+                returns complete frames up to that number. If the bytes do not
+                arrive prior to the 'timeout', an error is returned.
+
+                If 'timeout' is negative, this function waits indefinitely for
+                'number_of_bytes_to_read' frame bytes.
```

Comment:
We should probably document this as timeout is `nixnet.constants.TIMEOUT_INFINITE`